### PR TITLE
Implements support for automatically creating a zip archive of the pr…

### PR DESCRIPTION
**Pull request by viktorklang**
…otocols on package

Related to https://github.com/cloudstateio/cloudstate/issues/118
This would mean that we could have a checked in zip-archive which lang supports could fetch over HTTP and unzip.

### CODE REVIEW

**Comment by viktorklang** [comment]
Ping @marcellanz

**Comment by marcellanz** [comment]
> Ping @marcellanz

@viktorklang tested and works for me.

**Comment by viktorklang** [comment]
@marcellanz Cool, thanks for verifying! Should we include this in the developer documentation as well perhaps?

**Comment by marcellanz** [comment]
> @marcellanz Cool, thanks for verifying! Should we include this in the developer documentation as well perhaps?

@viktorklang sure :-)

**Comment by marcellanz** [comment]
@viktorklang examples was left out by intention right? Language support would need it to implement the TCK. "examples" are not part of the spec, but if the consumer if the zip is language support or alternative proxy implementers, the examples are needed by them. WDYT?

**Comment by viktorklang** [comment]
@marcellanz Yeah, I kept the shopping-cart protos outside of it. Do we want to have an extra zip-file for the shopping-cart protos?

**Comment by marcellanz** [comment]
@viktorklang I don't know, really. And name it tck-protos?

**Comment by viktorklang** [comment]
@marcellanz That's a great suggestion, we might have more of those later anyway. I'll amend this PR.

**Comment by viktorklang** [comment]
@marcellanz Check the latest commit(s)

**Comment by marcellanz** [comment]
@viktorklang 

works... should it get cleaned by `sbt:cloudstate> clean`?

Screenshot 2019-11-28 at 10 31 58

(digged into sbt documentation, let me know if it should get cleaned and I'll find out how, having to learn some sbt I think)

**Comment by viktorklang** [comment]
@marcellanz I'll add cleaning

**Comment by viktorklang** [comment]
@marcellanz Added the files to clean.

File: build.sbt [hunk 1]
```diff
@@ -99,19 +99,56 @@ headerSources in Compile ++= {
...
+        commonProtos
+      )
+
+      // Common TCK Language Support Proto Dependencies
+      IO.zip(archiveStructure("cloudstate-tck-protocols", base / "example" ** "*.proto"), tckProtos)
+
+      commonProtos
+    },
+    cleanFiles ++= Seq(
+        baseDirectory.value / (name.value + "-" + version.value + ".zip"),
```
Comment (marcellanz) [inline build.sbt]: @viktorklang this would miss the version if the branch is updated, ok for you?

Comment (viktorklang) [reply build.sbt]: @marcellanz I opted for being careful deleting files outside of the target directory. Perhaps we should not include the version number in the archive name at all?

Comment (marcellanz) [reply build.sbt]: >Perhaps we should not include the version number in the archive name at all?

@viktorklang That makes sense I think. One would get the zip like so perhaps:
```
curl -O https://raw.githubusercontent.com/cloudstateio/cloudstate/v0.5.0/protocols/cloudstate-protos.zip
```
where its clear which version one gets. Same for every other tag or branch.

Comment (viktorklang) [reply build.sbt]: :+1: Will amend.

Comment (viktorklang) [reply build.sbt]: @marcellanz Done! See latest commit

Comment (marcellanz) [reply build.sbt]: fetched and tested on my side, works! \o/

Comment (viktorklang) [reply build.sbt]: @marcellanz Awesome! :-)

**Comment by ihostage** [comment]
Hi, folks 👋 
@marcellanz When this PR will be merged, will you migrate a `go-support` to using these artifacts?

**Comment by marcellanz** [comment]
> Hi, folks 👋
> @marcellanz When this PR will be merged, will you migrate a `go-support` to using these artifacts?

@ihostage Its not a high priority yet, but yes I think user support libraries should get the protobuf files through these artefacts. Currently `go-support`has them, together with the `*.pb.go` files, distributed with a release. Do you have something in mind?

**Comment by ihostage** [comment]
> Do you have something in mind?
@marcellanz Nothing concrete 😄 
Just I wait for these artifacts and try to find time to play with another language support 😄
But the end of the year is a hard period and find time is very difficult. 😞

**Comment by marcellanz** [comment]
> > Do you have something in mind?
> 
> @marcellanz Nothing concrete 😄
> Just I wait for these artifacts and try to find time to play with another language support 😄
> But the end of the year is a hard period and find time is very difficult. 😞

@ihostage cool! Yeah time... I know, one can squeeze for a few weeks but then one have to pay tribute and compensate. Which language do you have in mind?

**Comment by viktorklang** [comment]
@marcellanz Merged! Also, added notes in https://github.com/cloudstateio/cloudstate/blob/master/docs/src/main/paradox/developer/language-support/index.md#creating-language-support-libraries

**Comment by marcellanz** [comment]
@viktorklang awesome, thanks!

**Pull request merged**